error.go

```diff
@@ -701,5 +701,12 @@ func (err WriteErrors) Count() int {
 }
 
 func (err WriteErrors) Error() string {
-	return fmt.Sprintf("kafka write errors (%d/%d)", err.Count(), len(err))
+	var sb strings.Builder
+	for i, writeError := range err {
+		if writeError == nil {
+			continue
+		}
+		fmt.Fprintf(&sb, "Message %d: %s \n", i, writeError.Error())
```

Comment:
Sorry, one list nit, it looks like the individual errors are at batch level not individual message level, `Message` here might be a bit misleading.

Reply:
@rhansen2 I removed the string builder part, instead of that I used a slice to be cleaner what do you think about it?

Reply:
Thanks @mhmtszr This all looks good to me!